source/features/clean-conversation-headers.tsx

```diff
@@ -1,68 +1,95 @@
-		add(byline) {
-			byline.classList.add('rgh-clean-conversation-header');
-			const {childNodes: bylineNodes} = byline;
-			// Removes: octocat opened this issue on 1 Jan [·] 1 comments
-			bylineNodes[4].textContent = bylineNodes[4].textContent!.replace('·', '');
-
-			// Removes: [octocat opened this issue on 1 Jan] · 1 comments
-			for (const node of [...bylineNodes].slice(0, 4)) {
-				node.remove();
-			}
-		}
-	});
-	deinit.push(observer.abort);
+async function initIssue(): Promise<void> {
+	const byline = await elementReady('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)');
+	if (!byline) {
+		return;
+	}
+
+	byline.classList.add('rgh-clean-conversation-header');
+
+	const {childNodes: bylineNodes} = byline;
+	// Removes: octocat opened this issue on 1 Jan [·] 1 comments
+	bylineNodes[4].textContent = bylineNodes[4].textContent!.replace('·', '');
+
+	// Removes: [octocat opened this issue on 1 Jan] · 1 comments
+	for (const node of [...bylineNodes].slice(0, 4)) {
+		node.remove();
+	}
 }
 
-function initPR(): void {
-	const observer = observe('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)', {
-		add(byline) {
-			byline.classList.add('rgh-clean-conversation-header');
-			const isSameAuthor = select('.js-discussion > .TimelineItem:first-child .author')?.textContent === select('.author', byline)!.textContent;
-			const baseBranch = select('.commit-ref.base-ref', byline)!;
-			const isDefaultBranch = (baseBranch.firstElementChild as HTMLAnchorElement).pathname.split('/').length === 3;
-
-			// Removes: [octocat wants to merge 1] commit into github:master from octocat:feature
-			// Removes: [octocat] merged 1 commit into master from feature
-			for (const node of [...byline.childNodes].slice(isSameAuthor ? 0 : 2, pageDetect.isMergedPR() ? 2 : 4)) {
-				node.remove();
-			}
-
-			baseBranch.previousSibling!.textContent = ' into ';
-			if (!isSameAuthor) {
-				byline.prepend('by ');
-			}
-
-			if (!isDefaultBranch && !(pageDetect.isClosedPR() && baseBranch.title.endsWith(':master'))) {
-				baseBranch.classList.add('rgh-clean-conversation-headers-non-default-branch');
-			}
-		}
-	});
-	deinit.push(observer.abort);
+function fromNodeNumber(): number {
+	if (pageDetect.isOpenPR()) {
+		return 9;
+	}
+
+	return pageDetect.isMergedPR() ? 5 : 7;
+}
+
+async function initPR(): Promise<void> {
+	const byline = await elementReady('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)');
+	if (!byline) {
+		return;
+	}
+
+	byline.classList.add('rgh-clean-conversation-header');
+
+	const author = select('.author', byline)!;
+	const isSameAuthor = pageDetect.isPRConversation() && author.textContent === (await elementReady('.TimelineItem .author'))!.textContent;
+
+	const base = select('.commit-ref', byline)!;
+	const baseBranch = base.title.split(':')[1];
+
+	// Replace the word "from" with an arrow
+	byline.childNodes[fromNodeNumber()].replaceWith(<> <ArrowLeftIcon/> </>);
```

Comment:
Just use `select` as previously asked. Probably
```suggestion
	base.nextSibling!.replaceWith(' ', <ArrowLeftIcon/>, ' ');
```

Reply:
I don't think so, but will double check

Reply:
Does not

Unless you want `base.nextSibling.nextSibling.nextSibling.nextSibling`

Reply:
$('.head-ref').previousSibling

Reply:
head-ref is not always there. can you look at the current code if its good?